File: chap19_retina/LM_stability.py

#  --------------------------------
#  Code to plot the stability curve for  the Longtin-Milton model of the pupil
#  light reflex.  

#  Used to generate the image in Fig. 26 of Chapter 19 
#  Keener and Sneyd, Mathematical Physiology, 3rd Edition, Springer.

#  Written by James Keener and James Sneyd
#  --------------------------------

import numpy as np
import matplotlib.pyplot as plt

eta = np.arange(np.pi / 2 + 0.01, np.pi, 0.01)
G = -1 / np.cos(eta)
delay = -eta / np.tan(eta)

plt.figure(figsize=(8, 6))
plt.plot(delay, G)
plt.axis([0, 10, 0, 10])
plt.xlabel('Delay')
plt.ylabel('Gain')
plt.text(1, 1, 'Stable', fontsize=18)
plt.text(4, 4, 'Unstable', fontsize=18)
plt.show()

File: chap19_retina/LM_stability.m

% --------------------------------
% Code to plot the stability curve for  the Longtin-Milton model of the pupil
% light reflex.  

% Used to generate the image in Fig. 26 of Chapter 19 
% Keener and Sneyd, Mathematical Physiology, 3rd Edition, Springer.

% Written by James Keener and James Sneyd
% --------------------------------

clear all
close all
clc
set(0,                           ...
'defaultaxesfontsize', 20,   ...
'defaultaxeslinewidth', 2.0, ...
'defaultlinelinewidth', 2.0, ...
'defaultpatchlinewidth', 0.7);

eta = [pi/2+.01:.01:pi];
G = -1./cos(eta);
delay = -eta./tan(eta);

figure(1)
    plot(delay,G)
    axis([0 10 0 10])
    xlabel('Delay')
    ylabel('Gain')
    box off
    text(1,1,'Stable','fontsize',18)
    text(4,4,'Unstable','fontsize',18)

    
    
